### ISSUE TITLE
Recordings screen out of memory error after switching from Away to Home

### ISSUE THREAD

**Issue by billmeek** [MythTV-Clients/MythTV-Android-Frontend#71]
This one is tough to duplicate, in other words, it doesn't happen
every time. The most reliable method I've found is:

<pre>
Reboot the Android device (to start on a clean system)
Start the application and select Home -> Recordings
Press every banner, top to bottom (I have 38)
Exit the Recordings screen and press Guide
Exit Guide and go back to Recordings
Press any recording, the following shows up in logcat:

11-06 23:46:55.000: E/dalvikvm-heap(2216): Out of memory on a 2720016-byte allocation.
</pre>

The entire log is here: http://paste.ubuntu.com/1339085/

This appears different from an earlier OOM case (which I didn't
report) that reliably failed when pressing the 34th recording, I
didn't have to exit the Recordings screen. I think this case was
fixed in: commit b92bf64 Fri Oct 26 12:54:55 2012 (attempt to
resolve oom loading coverarts.)

The above tested on: mythtv-android-frontend-1.0.2-76-gf29932c

**Comment by dmfrey** [comment]
@billmeek give it a shot now with the latest code.  I added some experiments in disk and image caching

**Comment by billmeek** [comment]
The whole application crashes when I enter the Recordings screen running on:
mythtv-android-frontend-1.0.2-77-gc4f6f0e. 

Details at: http://paste.ubuntu.com/1343209/

I may have been testing differently than you:

I've been having trouble starting the application from eclipse, so I almost always
have to go in the Android setup/applications menu and delete the cache/data and
application. Otherwise, I get eclipse errors, usually running out of memory.

**Comment by dmfrey** [comment]
let's chat about this.  The LogCat view in eclipse will filter all log
messages for the application.  I think we should start there with
addressing these issues.  This way anything else going on on the phone is
eliminated.

On Thu, Nov 8, 2012 at 1:37 PM, billmeek notifications@github.com wrote:

> The whole application crashes when I enter the Recordings screen running
> on:
> mythtv-android-frontend-1.0.2-77-gc4f6f0e.
> 
> Details at: http://paste.ubuntu.com/1343209/
> 
> I may have been testing differently than you:
> 
> I've been having trouble starting the application from eclipse, so I
> almost always
> have to go in the Android setup/applications menu and delete the
> cache/data and
> application. Otherwise, I get eclipse errors, usually running out of
> memory.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/MythTV-Clients/MythTV-Android-Frontend/issues/71#issuecomment-10199380.

## 

My Google Profile http://www.google.com/profiles/dmfrey

**Comment by billmeek** [comment]
To answer the above, filtering  on Application Name: org.mythtv. Let me know if you
want the filter changes: http://paste.ubuntu.com/1343292/

But I see something. Note that on line 93 in the above the IPv4 address ends in .204
(which is my Home 'real' backend, which was being tested.) On line 152 the address ends
in .223 which is my 'test' backend, always configured in my Away settings. I had been using
the Away backend before I switched to Home for this test.

With the Away backend off, all data removed and the Android rebooted, the "whole application crashes"
 with mythtv-android-frontend-1.0.2-77-gc4f6f0e isn't happening anymore.

And, I haven't been able to duplicate the original OOM issue.

I updated the subject lline.
